### DISCUSSION TITLE
Why TraverserContext variables use Class<?> as key?

### DISCUSSION THREAD

**Discussion by qzchenwl** [graphql-java/graphql-java#2721] [Q&A] [answered]
Why `TraverserContext` variables use `Class<?>` as key, why not just use `String`?
https://github.com/graphql-java/graphql-java/blob/0e6921e1ada072259c06dceb1dac105d03c1c515/src/main/java/graphql/util/TraverserContext.java#L127-L136

What if I have more than one variables of the same type?

Why `SchemaTraverser` is setting root var with different type(`SchemaTraverser.class` vs `Map<String, GraphQLNamedType>`) ?

https://github.com/graphql-java/graphql-java/blob/0e6921e1ada072259c06dceb1dac105d03c1c515/src/main/java/graphql/schema/SchemaTraverser.java#L41-L44

**Comment by bbakerman** [comment, marked as answer]
This was just to keep the API a little tighter.  Yes strings are excellent keys but they are also poor keys in the sense of knowing that a `CodeRegistry` is available as a var say.

So we chose class instead

> What if I have more than one variables of the same type?

So create a class and put those many things into that class

```
class ContainerOfThings {
   List<String,Object> things
}
```

You will know its your `ContainerOfThings` class!
